### PR TITLE
Adds Resource method to search metrics by type

### PR DESCRIPTION
**Pull request by josejulio**
Fetching a hawkular metric type id of a metric

### CODE REVIEW

File: spec/unit/inventory_spec.rb [hunk 1]
```diff
@@ -52,6 +66,31 @@
...
+
+    it 'top level metrics are of type Metric' do
+      expect(resource.metrics).to all(be_a(::Hawkular::Inventory::Metric))
+    end
+
+    it 'all metrics on the tree are of type Metric' do
+      expect(resource.metrics(true)).to all(be_a(::Hawkular::Inventory::Metric))
+    end
+
+    describe '#metrics_by_type' do
```
Comment (jotak) [inline spec/unit/inventory_spec.rb]: Is it intentional to put it inside `describe '#metrics'` block, rather than on the same level?

Comment (josejulio) [reply spec/unit/inventory_spec.rb]: No. My bad.

**Comment by jotak** [comment]
LGTM

**Comment by josejulio** [comment]
Thanks @israel-hdez

**Pull request merged**